app/tlsconfig.go

```diff
@@ -0,0 +1,52 @@
+package app
+
+import (
+	"crypto/tls"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+)
+
+func getTLSConfig() (*tls.Config, error) {
```

Comment:
Can we add comments in here? It's difficult to follow what these values + cases + functions mean/are doing without already being knowledgable in the area